allennlp/models/ensemble/bidaf_ensemble.py

```diff
@@ -0,0 +1,158 @@
+from typing import Dict, List, Any, Tuple
+
+from overrides import overrides
+import torch
+
+from allennlp.models.ensemble.ensemble import Ensemble
+from allennlp.models.archival import load_archive
+from allennlp.models.model import Model
+from allennlp.models.reading_comprehension import BidirectionalAttentionFlow
+from allennlp.common import Params
+from allennlp.data import Vocabulary
+from allennlp.training.metrics import SquadEmAndF1
+
+@Model.register("bidaf-ensemble")
+class BidafEnsemble(Ensemble):
+    """
+    This class ensembles the output from multiple BiDAF models.
+
+    It combines results from the submodels by taking the option with the most votes and breaking ties with
+    the average confidence of the start and stop indices.
+    """
+
+    def __init__(self,
+                 submodels: List[BidirectionalAttentionFlow]) -> None:
+        super(BidafEnsemble, self).__init__(submodels)
+
+        # Using ModuleList propagates calls to .eval() so dropout is disabled on the submodels in evaluation
+        # and prediction.
+        self.submodels = torch.nn.ModuleList(submodels)
```

Comment:
I think this should just be in the base `Ensemble` constructor (you are already doing `self.submodels = submodels`) there

Reply:
Good point--in particular it shouldn't be defined in both places.